docs/developer-docs/integrations/sns/managing-sns/sns-asset-canister.md

```diff
@@ -0,0 +1,146 @@
+:::info 
+To deploy to the mainnet network, you will need a wallet that contains cycles. For more information on cycles wallets, please see [here](../../../setup/cycles/cycles-wallet.md).
+:::
+
+To deploy your asset canister to the mainnet, the following command can be used:
+
+```
+dfx deploy assets --network "ic" --wallet <principal>
+```
+
+Once the asset canister has been deployed, the SNS can be launched. You can learn more about launching an SNS [here](../launch-sns/launch-sns.md)
+
+## Submitting an SNS proposal and upgrading an asset canister
+
+- #### Step 1: First, upgrade the asset canister (by proposal, as with any other canister) to the asset canister wasm bundled with dfx 0.14.1, available [here](https://github.com/dfinity/sdk/blob/release-0.14.1/src/distributed/assetstorage.wasm.gz).
+
+- #### Step 2: After upgrading the asset canister by proposal, have someone with `Prepare` permission run:
+
+```
+dfx deploy <frontend canister name> --network ic --by-proposal
+```
+
+The output will contain something like this:
+
+```
+Proposed commit of batch 2 with evidence e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855. Either commit it by proposal, or delete it.
+```
+
+- #### Step 3: Save the batch number and the evidence value for use with the asset canister API.
+
+- #### Step 4: To ensure that others would be able to verify the evidence in the proposal, have someone else clone the dapp repo and run:
+
+```
+dfx deploy <frontend canister name> --network ic --compute-evidence
+```
+
+The computed evidence should match the evidence from step 2.
+
+- #### Step 5: Submit a new proposal to commit the batch, using the APIs below:
+
+Use these asset canister APIs in the proposal:
+
+```
+   type CommitProposedBatchArguments = record {
+   batch_id: BatchId;
+   evidence: blob;
+  };
+  type ValidationResult = variant { Ok : text; Err : text };
+
+
+    validate_commit_proposed_batch: (CommitProposedBatchArguments) -> (ValidationResult);
+  commit_proposed_batch: (CommitProposedBatchArguments) -> ();
+
+```
+
+An example can be found [here](https://github.com/dfinity/sdk/blob/master/e2e/tests-dfx/assetscanister.bash#L133).
+
+If the proposal is rejected, the preparer should use this new asset canister API:
+
+```
+  type DeleteBatchArguments = record {
+    batch_id: BatchId;
+  };
+  delete_batch: (DeleteBatchArguments) -> ();
+```
+
+
+## Asset canister examples
+
+An example of an SNS asset canister is canister `sqbzf-5aaaa-aaaam-aavya-cai`, which is an asset canister part of the [Dragginz Dapp SNS](https://dashboard.internetcomputer.org/canister/sqbzf-5aaaa-aaaam-aavya-cai).
+
+## SNS GenericNervousSystemFunctions
+
+To submit a new `AddGenericNervouSystemFunction` SNS Proposal to support the `commit_proposed_batch` API, the target canister id should be the asset canister (that is upgraded in upgrade steps) and the target function is `commit_proposed_batch`. The validate function should be `validate_commit_proposed_batch`. 
```

Comment:
I think this needs to happen before the above upgrade can work. Therefore I would maybe restructure this sligthly so that it is in order of the actions that a dev has to do. WDYT?